### ISSUE TITLE
PDF search not working for web

### ISSUE THREAD

**Issue by hmbadhon** [syncfusion/flutter-widgets#1047]
@deepika134
Pdf search not working in flutter web

sample code:

SfPdfViewer.network(
pdfModel!.fileUrl!,
controller: _pdfViewerController,
enableDoubleTapZooming: false,
canShowScrollStatus: false,
canShowPaginationDialog: false,
canShowScrollHead: false,
enableTextSelection: false,
enableDocumentLinkAnnotation: false,
canShowHyperlinkDialog: false,
enableHyperlinkNavigation: false,
canShowPasswordDialog: false,
onDocumentLoaded: (details) {
setState(() {});
},
onDocumentLoadFailed: (failed) {
setState(() {
isPdfLoadFailed = true;
});
},
onPageChanged: (callback) {
setState(() {});
},
),

see screen recoding browser got stack :

https://user-images.githubusercontent.com/24831033/213963138-d9f9e59d-3ba0-4cf3-a13e-727809a96a3e.mp4

**Comment by hmbadhon** [comment]
https://prnt.sc/mot2gY6AfKy-

**Comment by deepika134** [comment]
Hi @hmbadhon ,With the given information, we tried to reproduce the reported issue, but on our end, text search works as expected. For your reference, we have prepared a sample. You can download the sample and sample output screenshot from the attachment. If you are still able to reproduce the reported issue in your use case, kindly share the modified issue reproducing sample with us.
[sample.zip](https://github.com/syncfusion/flutter-widgets/files/10486285/sample.zip)

**Comment by hmbadhon** [comment]
@deepika134 can you try it with network asset and also search through input field with big pdf asset?

**Comment by hmbadhon** [comment]
@deepika134 check the sample bug file, where i regenarate the issue. thanks
[github_1047_sample_bug.zip](https://github.com/syncfusion/flutter-widgets/files/10486406/github_1047_sample_bug.zip)

**Comment by deepika134** [comment]
Hi @hmbadhon ,we have resolved the Text Search not working issue in your sample. The modified sample can be downloaded from the attachment. But, when we tested the sample, we found the issue Highlighted instance bounds is not proper in the provided document and the fix for this issue will be included in our upcoming weekly release which will be available on February 7, 2022.Use the below feedback link to track the status of the reported bug. https://www.syncfusion.com/feedback/40664/find-text-is-not-working-properly-in-specific-pdf-document
Disclaimer: “Inclusion of this solution in the weekly release may change due to other factors including but not limited to QA checks and works reprioritization.” 
[github_1047_sample_bug.zip](https://github.com/syncfusion/flutter-widgets/files/10519147/github_1047_sample_bug.zip)

**Comment by hmbadhon** [comment]
@deepika134 thanks for the solution but i need the search onChange state but you solved it with onSubmit. Will this gonna solve in next update?

**Comment by deepika134** [comment]
Hi @hmbadhon , We have modified the sample as per your requirement and it can be downloaded from the attachment. Kindly let us know if you have any concerns.
[github_1047_sample.zip](https://github.com/syncfusion/flutter-widgets/files/10553659/github_1047_sample.zip)

**Comment by hmbadhon** [comment]
@deepika134 thanks for the support but by changing textfield onChange it freeze on web and load after some time like i saw you on the video. can i get the fix on next update?

**Comment by deepika134** [comment]
Hi @hmbadhon , In the web platform, the search will be performed synchronously and so the result will be returned only after completing the search on all the pages. This is because flutter does not supported isolates for the web platform yet. Please see  the following link for your reference , https://api.flutter.dev/flutter/dart-isolate/dart-isolate-library.html

**Comment by kazimdsaidul** [comment]
@deepika134 , Can we use isolated_worker alternate solution of isolate?

https://pub.dev/packages/isolated_worker

**Comment by deepika134** [comment]
Hi @hmbadhon , we have included the fix for this issue " Find text is not working properly in specific PDF document" in our latest weekly NuGet release, so kindly upgrade to the latest version **v20.4.50** to get the issue resolved. 

Package link: https://pub.dev/packages/syncfusion_flutter_pdfviewer/versions/20.4.50

**Comment by hmbadhon** [comment]
Hi @deepika134, Still not working pdf search properly for big pdf like 100+ pages in flutter web.

**Comment by deepika134** [comment]
Hi @hmbadhon , With the given information, we tried to reproduce the reported issue, but on our end, text search works as expected. For your reference, we have prepared a sample. You can download the sample and sample output screen record from the attachment. If you are still able to reproduce the reported issue in your use case, kindly share the modified issue reproducing sample with us.
[textsearch_1047.zip](https://github.com/syncfusion/flutter-widgets/files/10836446/textsearch_1047.zip)

**Comment by hmbadhon** [comment]
@deepika134 check the sample bug file, where i regenerate the issue for flutter web. thanks
[github_1047_sample_bug.zip](https://github.com/syncfusion/flutter-widgets/files/10836525/github_1047_sample_bug.zip)

**Comment by deepika134** [comment]
Hi @hmbadhon , As we mentioned earlier in web platform, the search will be performed synchronously. When the text search is initiated, we do not perform any other operation in the PDFViewer and also the result will be returned only after completing the search on all the pages. I have checked the sample you shared on text search and it works fine on our side and I have attached its output screen recording for your reference.
[textsearchoutput.zip](https://github.com/syncfusion/flutter-widgets/files/10856631/textsearchoutput.zip)

**Comment by deepika134** [comment]
Hi @hmbadhon , As we mentioned earlier in web platform, the search will be performed synchronously. When the text search is initiated, we do not perform any other operation in the PDFViewer and also the result will be returned only after completing the search on all the pages. I have checked the sample you shared on text search and it works fine on our side and I have attached its output screen recording for your reference.
[textsearchoutput.zip](https://github.com/syncfusion/flutter-widgets/files/10856631/textsearchoutput.zip)
